core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -1062,6 +1062,11 @@ public String getMessage() {
   public static final String KETTLE_GLOBAL_LOG_VARIABLES_CLEAR_ON_EXPORT = "KETTLE_GLOBAL_LOG_VARIABLES_CLEAR_ON_EXPORT";
 
   /**
+   * Set this variable to true to disable receiving file names from another step for XML Input Stream (StAX) step.
+   */
+  public static final String KETTLE_IGNORE_RECEIVING_INPUT = "KETTLE_IGNORE_RECEIVING_INPUT";
```

Comment:
Strelchanka, I don't like the change done this way because it becomes an all or nothing enterprise (you always ignore the receiving input or you don't). Instead, I think I'd rather see the step behave as it did before if there's already a value in the field irrespective of whether there's an incoming hop.